nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/WorkManagerImpl.java

```diff
@@ -678,17 +691,21 @@ public void shutdownAndSuspend() throws InterruptedException {
                 deactivateQueueMetrics(queueId);
                 queuing.setActive(queueId, false);
                 // suspend all running work
+                boolean hasWorkRunning = false;
                 for (Work work : running) {
                     work.setWorkInstanceSuspending();
                     log.trace("suspending and rescheduling " + work.getId());
                     work.setWorkInstanceState(State.SCHEDULED);
                     queuing.workReschedule(queueId, work);
+                    hasWorkRunning = true;
                 }
-                // sleep for a given amount of time for works to have time to persist their state and stop properly
-                long shutDowndelay = Long.parseLong(Framework.getService(ConfigurationService.class)
-                                                             .getProperty(SHUTDOWN_DELAY_KEY, "0"));
-                if (shutDowndelay > 0) {
-                    Thread.sleep(shutDowndelay);
+                if (hasWorkRunning) {
+                    // sleep for a given amount of time for works to have time to persist their state and stop properly
+                    long shutdownDelay = Long.parseLong(Framework.getService(ConfigurationService.class)
+                                                                 .getProperty(SHUTDOWN_DELAY_MS_KEY, "0"));
+                    if (shutdownDelay > 0) {
+                        Thread.sleep(shutdownDelay);
```

Comment:
I didn't review the code that originally added this, but we could be a bit nicer: in a loop sleep 100ms and re-check if there are still running works, until there are no more or the complete timeout has expired. This allows for fast-stop once all works are finished.